python/revng/internal/api/manager.py

```diff
@@ -52,8 +66,14 @@ def __init__(
     def uid(self) -> int:
         return int(ffi.cast("uintptr_t", self._manager))
 
+    def _save_hook(self):
+        result = self.synchronizer.save(self.workdir)
+        assert result, "Failed to save manager"
+
     def save(self) -> bool:
```

Comment:
Document the fact that this method might be called by multiple threads and that `self.synchronizer.save` is designed to run in parallel.

Reply:
Added a docstring